### PR TITLE
Continued stabilization for the SSH policy executed by the Universal NRP and the Security Baseline module

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

Continued stabilization for the SSH policy executed by the Universal NRP and the Security Baseline module, including bug fixes and improvements for the case when the SSH server is not installed on the device.

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/common/commonutils/SshUtils.c [hunk 1]
```diff
@@ -473,37 +479,41 @@ static int CheckSshWarningBanner(const char* bannerFile, const char* bannerText,
...
+            {
+                OsConfigLogError(log, "CheckSshWarningBanner: cannot read from '%s'", bannerFile);
+                OsConfigCaptureReason(reason, "'%s' is set to '%s' but the file cannot be read",
+                    "%s, also '%s' is set to '%s' but the file cannot be read", banner, actualValue);
+                status = ENOENT;
+            }
+            else  if (0 != strcmp(contents, bannerText))
+            {
+                OsConfigLogError(log, "CheckSshWarningBanner: banner text is:\n%s instead of:\n%s", contents, bannerText);
+                OsConfigCaptureReason(reason, "Banner text from file '%s'is different from not the expected text",
```
Comment (simonjaeger) [inline src/common/commonutils/SshUtils.c:503]: nit: Spacing between `'%s'` and `is`

**Review by simonjaeger — APPROVED** [review]
LGTM

File: src/common/commonutils/SshUtils.c (hunk 1 above)
Comment (MariusNi) [reply src/common/commonutils/SshUtils.c:503]: Good catch!

**Review by AhmedBM — APPROVED** [review]
👌

**Review by simonjaeger — APPROVED** [review]
LGTM

**Pull request merged**